tests/unit/components/status/UploadFilesWidgets.spec.js

```diff
@@ -0,0 +1,100 @@
+import Vuex from "vuex";
+import { createLocalVue } from "@vue/test-utils";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+let NuxtStore;
+let store;
+
+describe("UploadFilesWidget.vue", () => {
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => wrapper.destroy());
+  test("When mounting UploadFilesWidget from the build dist file, Then it loads successfully and the background black box is displayed", () => {
+    const propsData = {};
+    wrapper = shallowMount(dist_UploadFilesWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_background_div = wrapper.find(".div__upload-file-background");
+    expect(target_background_div.isVisible()).toBe(true);
+  });
+  test("When mounting UploadFilesWidget, Then it loads the 'files uploaded' and 'total files to upload' is as specified in the Vuex values in settings file_count and max_file_count respectively", () => {
+    const uploaded_files = 500;
+    const total_files_to_upload = 900;
+
+    store.commit("settings/set_file_count", uploaded_files);
+    store.commit("settings/set_max_file_count", total_files_to_upload);
+
+    const propsData = {};
+    wrapper = mount(UploadFilesWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+
+    const target_upload_file_count_container = wrapper.find(
+      ".span__upload-file-count-container"
+    );
+    expect(target_upload_file_count_container.text()).toStrictEqual("500/900");
+  });
+  test("Given that the uploaded_files is equal to max_file_count, When complete upload is successful, Then the 'check mark' is visible indicating the completion of file upload activity", async () => {
+    const propsData = {};
+    wrapper = mount(UploadFilesWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const uploaded_files = 900;
+    const total_files_to_upload = 900;
+
+    store.commit("settings/set_file_count", uploaded_files);
+    store.commit("settings/set_max_file_count", total_files_to_upload);
+
+    await wrapper.vm.$nextTick();
+    const target_upload_custom_check_mark = wrapper.find(
+      ".div__upload-custom-check-mark"
+    );
+    expect(target_upload_custom_check_mark.isVisible()).toBe(true);
+    const target_upload_custom_check_mark_color = wrapper.find(
+      ".svg-inline-fa-check-path"
+    );
+    expect(
+      target_upload_custom_check_mark_color.attributes("fill")
+    ).toStrictEqual("#00c46f");
+  });
+  test("Given that the uploaded_files is less than max_file_count, When the upload of files is about 88%, Then the 'progress bar' is visible indicating and its less than the max 110px max length of progress bar attribute", async () => {
```

Comment:
In general, if Vuex commits happen prior to mounting the component, then they are part of the Arrange/Given section of a test.  Here it looks like you are testing the behavior of the component When it mounts

Reply:
Corrected, to have Vuex commit to happen after mount ? assume if one experiment produced 500 files and was uploaded, then the device was re-set and the old files are present in the laptop/computer's persistence storage. and an new experiment produces 400 files then would you start from 500 to 900 on the App or would you reset to start from 0 or would it start Vuex as 500 to 900 ?